src/vm.c

```diff
@@ -258,23 +258,26 @@ static void uv__prepare_cb(uv_prepare_t *handle) {
     uv__maybe_idle(qrt);
 }
 
-static void uv__check_cb(uv_check_t *handle) {
-    TJSRuntime *qrt = handle->data;
-    CHECK_NOT_NULL(qrt);
-
-    JSRuntime *rt = qrt->rt;
-    JSContext *ctx1;
+void tjs_execute_jobs(JSContext *ctx) {
+    JSRuntime *rt = JS_GetRuntime(ctx);
     int err;
 
     /* execute the pending jobs */
     for (;;) {
-        err = JS_ExecutePendingJob(rt, &ctx1);
+        err = JS_ExecutePendingJob(rt, &ctx);
```

Comment:
THis is not correct. Please leave ctx1 and take a TJSRuntime as an argument to tjs_execute_jobs. The reason is that JS_ExecutePendingJob returns the context that is resumed in the passed ctx argument. It need not be the one we have (if we ever go multi-context).

Reply:
I can't get the `TJSRuntime` directly in the timer's callback. Current solution aligns with the `js_std_loop` pattern in QuickJS.